cudaextender/include/claraparabricks/genomeworks/cudaextender/extender.hpp

```diff
@@ -0,0 +1,151 @@
+{
+    return ((x.seed_pair.target_position_in_read == y.seed_pair.target_position_in_read) && (x.seed_pair.query_position_in_read == y.seed_pair.query_position_in_read) && (x.length == y.length) && (x.score == y.score));
+}
+
+/// CUDA Extender Class
+class Extender
+{
+public:
+    /// \brief Virtual destructor for Extender.
+    virtual ~Extender() = default;
+
+    /// \brief Host pointer API for extension
+    ///
+    /// Takes values from host data structures, copies them over to device,
+    /// launches async extension kernels on specified stream. Filters
+    /// segments on device based on score_threshold
+    ///
+    /// \param[in] h_query Host pointer to encoded query sequence
+    /// \param[in] query_length Length of query sequence
+    /// \param[in] h_target Host pointer to encoded target sequence
+    /// \param[in] target_length Length of target sequence
+    /// \param[in] score_threshold Score threshold for filtering extended segments
+    /// \param[in] h_seed_pairs Vector of seed pairs mapping to query and target
+    ///                         sequences
+    /// \return Status of the async extension launch
+    virtual StatusType extend_async(const char* h_query, const int32_t& query_length,
+                                    const char* h_target, const int32_t& target_length,
+                                    const int32_t& score_threshold,
+                                    const std::vector<SeedPair>& h_seed_pairs) = 0;
+
+    /// \brief Device pointer API for extension
+    ///
+    /// All inputs to this function are expected to be resident on the device
+    /// before launch. Output array d_scored_segment_pairs must be pre-allocated on device.
+    /// Launches async extension kernel. Filters segments on device based on score_threshold.
+    /// The stream associated with this function call must be synchronized to before accessing
+    /// any output parameters.
+    ///
+    /// \param[in] d_query Device pointer to encoded query sequence
+    /// \param[in] query_length Length of query sequence
+    /// \param[in] d_target Device pointer to encoded target sequence
+    /// \param[in] target_length Length of target sequence
+    /// \param[in] score_threshold Score threshold for filtering extended segments
+    /// \param[in] d_seed_pairs Device pointer to array of seed pairs mapping between
+    ///                         target and query sequences
+    /// \param[in] num_seed_pairs Length of d_seed_pairs array
+    /// \param[out] d_scored_segment_pairs Pointer to a pre-allocated device location for
+    ///                                    storing extension output
+    /// \param[out] d_num_scored_segment_pairs Pointer to pre-allocated device location for
+    ///                                        storing length of extension output
+    /// \return Status of the async extension launch
+    virtual StatusType extend_async(const char* d_query, int32_t query_length,
```

Comment:
I'm wondering now perhaps if this should be a `char*` or `int8_t/uint8_t*`? @ahehn-nv @atadkase

Reply:
I think that this will change based on the encoding API that @ahehn-nv will design. It can be made int8_t*.

Reply:
Yes, I think it makes sense to change this to `int8_t*`. That clarifies that it the API expects a sequence of numeric values and not a character string.

Reply:
Done.